pyspecProcScripts/simple_functions.py

```diff
@@ -127,10 +127,42 @@ def determine_sign(s, direct="t2", fl=None):
     if fl is not None:
         fl.next("check sign")
         if "vd" in s.dimlabels:
-            fl.image(
-                s.C.setaxis("vd", "#").set_units("vd", "scan #") * data_sgn
-            )
+            fl.image(s.C.setaxis("vd", "#").set_units("vd", "scan #") * data_sgn)
         else:
             fl.image(s * data_sgn)
         fl.pop_marker()
     return data_sgn
+
+
+def clock_correction(s, axis_along, direct="t2", max_cyc=0.5):
+    """
+    Parameters
+    ==========
+    s: nddata
+       data to be corrected
+    axis_along: str
+                indirect axis to apply the clock
+                correction to
+    direct: str
+            direct axis
+    max_cyc: float
+             maximum expected cycles per s of the dataset
+    Return
+    ======
+    final_corr: float
+                clock corrected shift
+    """
+    for_correct = s.C
+    Delta = np.diff(s[axis_along][np.r_[0, -1]]).item()
+    correction_axis = psd.nddata(np.r_[-0.5:0.5:300j] * max_cyc / Delta, "correction")
+    for_correct = for_correct * np.exp(
+        -1j * 2 * np.pi * correction_axis * for_correct.fromaxis(axis_along)
+    )
+    # {{{ determine the best sign flip for each correction
```

Comment:
this I pulled from JF notes for processing nutation [here](https://jmfrancklab.slack.com/archives/DKTRA59D2/p1721317867955689?thread_ts=1721316236.555979&cid=DKTRA59D2) and this works well